interp/src/primitives/combinational.rs

```diff
@@ -0,0 +1,164 @@
+use super::{stateful::get_param, Primitive};
+use crate::comb_primitive;
+use crate::values::Value;
+use calyx::ir;
+use std::ops::Not;
+
+/// A constant.
+#[derive(Default, Debug)]
+pub struct StdConst {
+    value: Value,
+}
+
+impl StdConst {
+    pub fn from_constants(value: u64, width: u64) -> Self {
+        StdConst {
+            value: Value::from(value, width).unwrap(),
+        }
+    }
+
+    pub fn new(params: ir::Binding) -> Self {
```

Comment:
what's this `ir::Binding` thing? I'm not sure I've seen it before, is it new?

Reply:
You can find the definition in the calyx portion of the repo, but it's just a smallvec tuple containing parameter Ids associated with their u64 values.